compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -646,18 +646,20 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             }
         }
 
-        // Confirm the `type Output: Sized;` bound that is present on `FnOnce`
-        let output_ty = self.infcx.instantiate_binder_with_placeholders(sig.output());
-        let output_ty = normalize_with_depth_to(
-            self,
-            obligation.param_env,
-            cause.clone(),
-            obligation.recursion_depth,
-            output_ty,
-            &mut nested,
-        );
-        let tr = ty::TraitRef::from_lang_item(self.tcx(), LangItem::Sized, cause.span, [output_ty]);
-        nested.push(Obligation::new(self.infcx.tcx, cause, obligation.param_env, tr));
+        // Confirm the `Sized` bounds that are present on the argument
```

Comment:
Do we need to do this during confirmation? I think we could probably just register `Sized` obligations when checking a closure during HIR typeck.

Reply:
Now that I think more about this, I'm not sure there is a way to actually hit this check—the `Sized` bound on the type parameter is checked anyway, to ensure the trait is well-formed. I think I will just remove it

Reply:
Wait, this definitely still ICEs.

```
#![feature(unsized_fn_params)]

fn bad() -> fn([u8]) { todo!() }

fn main() {
    let f: fn([u8]) = |_| {};
    let slice: Box<[u8]> = Box::new([1; 8]);
    f(*slice);
}
```

Reply:
> Wait, this definitely still ICEs.

Not on my machine...

Reply:
... I didn't copy and paste that example correctly